google/cloud/testing_util/validate_metadata.cc

```diff
@@ -425,6 +372,51 @@ void ValidateMetadataFixture::IsContextMDValid(
   EXPECT_THAT(actual, UnorderedElementsAreArray(matchers));
 }
 
+void ValidateMetadataFixture::ExchangeMetadata(
+    grpc::ClientContext& client_context,
+    grpc::GenericServerContext& server_context) {
+  void* tag;
+  bool ok;
+
+  // Create an in-process stub to make client calls with.
+  grpc::GenericStub generic_stub(
+      server_->InProcessChannel(grpc::ChannelArguments()));
+
+  // Start a call, the contents of which, do not matter.
+  auto cli_stream =
+      generic_stub.PrepareCall(&client_context, "made_up_method", &cli_cq_);
+  auto call_tag = std::make_unique<int>();
+  cli_stream->StartCall(call_tag.get());
+  cli_cq_.Next(&tag, &ok);
+  ASSERT_TRUE(ok && tag == call_tag.get()) << "stub.PrepareCall() failed.";
```

Comment:
Are the heap allocations necessary, or could we just use `int call_tag`?

Reply:
the latter. https://github.com/googleapis/google-cloud-cpp/pull/13280